syntax = "proto3";

option objc_class_prefix = "LSS";

package login_system;

// The greeting service definition.
service LoginSystem {
	//test hello
	rpc hello (helloRequest) returns (helloResponse) {}
	//注册接口
	rpc registerAccount (registerRequest) returns (registerResponse) {}
	//登录接口
	rpc loginAccount (loginRequest) returns (stream loginResponse) {}
	//session ticket 校验接口
	rpc verifyST (verifySTRequest) returns (verifySTResponse) {}
}

message helloRequest {
	string helloClient = 1;
}
message helloResponse {
	string helloServer = 1;
}

message registerRequest {
	string hash = 1;
	string salt = 2;
	string nickname = 3;
	string phone_num = 4;
}
message registerResponse {
	int32 ret = 1;
	string msg = 2;
	string user_id = 3;
}

message loginRequest {
	string user_id = 1;
	string data = 2;
}
message loginResponse {
	int32 ret = 1;
	string msg = 2;
 	string st = 3;
 	bool logout = 4;
}

message verifySTRequest {
	string user_id = 1;
	string st = 2;
}
message verifySTResponse {
	int32 ret = 1;
	string msg = 2;
}